app/javascript/components/forms/firmware-update.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from "react";
+import { Form, Field, FormSpy } from "react-final-form";
+import { Form as PfForm, Grid, Button, Col, Row, Spinner } from "patternfly-react";
+import PropTypes from "prop-types";
+import { required } from "redux-form-validators";
+
+import { FinalFormField, FinalFormSelect } from "@manageiq/react-ui-components/dist/forms";
+import '@manageiq/react-ui-components/dist/forms.css';
+
+const FirmwareUpdate = ({loading, updateFormState, physicalServerIds, initialValues, firmwareBinaries}) => {
+  if(loading){
+    return (
+      <Spinner loading size="lg" />
+    );
+  }
+
+  return (
+    <Form
+      onSubmit={() => {}} // handled by modal
+      initialValues={initialValues}
+      render={({ handleSubmit }) => (
+        <PfForm horizontal>
+          <FormSpy onChange={state => updateFormState({ ...state, values: state.values })} />
```

Comment:
I don't think you need the spread operator here (i think that is a left over of our debugging) just pas the state to `updateFormState`

Reply:
Ha